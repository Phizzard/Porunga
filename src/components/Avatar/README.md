```jsx
<Avatar text="KB" round={false} />
```

```jsx
<Avatar imgSrc="./porunga-close.png" size="large" />
```
