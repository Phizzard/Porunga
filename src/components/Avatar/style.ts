import styled from '@emotion/styled'

interface AvatarProps {
  size?: string
  round?: boolean
}

interface AvatarTextProps {
  bgColor?: string
  color?: string
  size?: string
  round?: boolean
}

export const AvatarImage = styled.img<AvatarProps>`
  height: ${p => p.size};
  width: ${p => p.size};
  object-fit: contain;
  border-radius: ${p => (p.round ? '50%' : 'none')};
`

export const AvatarText = styled.span<AvatarTextProps>`
  display: flex;
  justify-content: center;
  align-items: center;
  font-family: Open Sans, sans-serif;
  background-color: ${p => p.bgColor};
  color: ${p => p.color};
  height: ${p => p.size};
  width: ${p => p.size};
  border-radius: ${p => (p.round ? '50%' : 'none')};
  font-size: 1.5rem;
`
