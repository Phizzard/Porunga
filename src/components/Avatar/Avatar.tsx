import React from 'react'
import { useTheme } from 'emotion-theming'

import { ThemeConfig, Sizes, Palette } from '../../types'
import { get } from '../../utils/get'
import { AvatarImage, AvatarText } from './style'

export interface AvatarProps {
  imgSrc?: string
  size?: Sizes
  bgColor?: Palette
  textColor?: Palette
  text?: string
  alt?: string
  round?: boolean
  onClick?: React.MouseEventHandler<HTMLDivElement>
  containerProps?: any
  props?: any
}

export const Avatar = ({
  imgSrc,
  size = 'medium',
  bgColor = 'brand',
  textColor = 'light-1',
  text,
  round = true,
  alt = '',
  onClick,
  ...props
}: AvatarProps) => {
  const theme: ThemeConfig = useTheme()

  const avatarProps = {
    size: get(theme, 'avatar', 'sizes', size),
    round,
    ...props,
  }

  const avatarTextProps = {
    bgColor: get(theme, 'root', 'colors', bgColor),
    color: get(theme, 'root', 'colors', textColor),
    ...avatarProps,
  }

  return (
    <div onClick={onClick}>
      {imgSrc ? (
        <AvatarImage src={imgSrc} alt={alt} {...avatarProps} />
      ) : (
        <AvatarText {...avatarTextProps}>{text}</AvatarText>
      )}
    </div>
  )
}

export default Avatar
