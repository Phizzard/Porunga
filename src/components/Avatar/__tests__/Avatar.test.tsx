import React from 'react'
import renderer from 'react-test-renderer'
import Avatar from '../Avatar'

it('renders correctly', () => {
  const component = renderer.create(<Avatar text="KB" round={false} />)

  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
