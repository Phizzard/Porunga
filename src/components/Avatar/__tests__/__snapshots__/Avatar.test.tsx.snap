// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <span
    className="css-1x4lckw"
  >
    KB
  </span>
</div>
`;
